uaclient/cli.py

```diff
@@ -1009,6 +1037,75 @@ def action_attach(args, *, cfg):
     )
 
 
+def _write_command_output_to_file(
+    cmd, filename: str, return_codes: List[int] = None
+) -> None:
+    """Helper which runs a command and writes output or error to filename."""
+    try:
+        out, _ = util.subp(cmd.split(), rcs=return_codes)
+    except util.ProcessExecutionError as e:
+        util.write_file("{}-error".format(filename), str(e))
+    else:
+        util.write_file(filename, out)
+
+
+def _collect_file(path: str, out_dir: str) -> None:
+    if os.path.isfile(path):
+        shutil.copy(path, out_dir)
+
+
+# We have to assert root here, because the logs are not non-root user readable
+@assert_root
+def action_collect_logs(args, *, cfg: config.UAConfig):
+    output_file = args.output or UA_COLLECT_LOGS_FILE
+
+    with tempfile.TemporaryDirectory() as output_dir:
+
+        _write_command_output_to_file(
+            "cloud-id", "{}/cloud-id.txt".format(output_dir)
+        )
+        _write_command_output_to_file(
+            "ua status --format json", "{}/ua-status.json".format(output_dir)
+        )
+        _write_command_output_to_file(
+            "canonical-livepatch status",
+            "{}/livepatch-status.txt".format(output_dir),
+        )
+        _write_command_output_to_file(
+            "systemctl list-timers --all",
+            "{}/systemd-timers.txt".format(output_dir),
+        )
+        _write_command_output_to_file(
+            "journalctl --boot=0 -o short-precise",
+            "{}/journalctl.txt".format(output_dir),
+        )
+        _write_command_output_to_file(
+            "dmesg", "{}/dmesg.txt".format(output_dir)
+        )
+
+        for service in UA_SERVICES:
+            _write_command_output_to_file(
+                "systemctl status {}".format(service),
+                "{}/{}.txt".format(output_dir, service),
+                return_codes=[0, 3],
+            )
+
+        ua_logs = (
+            cfg.cfg_path,
+            cfg.log_file,
+            cfg.timer_log_file,
+            cfg.license_check_log_file,
+            cfg.data_path("jobs-status"),
+            "/etc/apt/sources.list.d/ubuntu*",
```

Comment:
Also, this one check here is a little bit generic.
We could potentially take more files than needed (which is not a problem per se)
@blackboxsw @orndorffgrant do you think we should only collect logs for the `RepoEntitlements` entries ?

Reply:
I see the point,
Please let me know if this may be problematic, so we can check for specific files.

Reply:
Done - checking only for the specific `RepoEntitlements` we have